### ISSUE TITLE
Selecting hidden difficulties on a non-standard mode reverts it back to standard mode

### ISSUE THREAD

**Issue by AnzoP** [ppy/osu#9032]
Current setting on my end is: Display Beatmap from 4 starts.
When selecting a hidden difficulty, mode reverts back to standard.

Screenshots or videos showing encountered issue:
https://youtu.be/N6RWNlD2HKU

osu!lazer version:
2020.508.2

Logs:
[performance.log](https://github.com/ppy/osu/files/4633296/performance.log)
[runtime.log](https://github.com/ppy/osu/files/4633297/runtime.log)
[updater.log](https://github.com/ppy/osu/files/4633298/updater.log)
[database.log](https://github.com/ppy/osu/files/4633300/database.log)

**Comment by smoogipoo** [comment]
Related to https://github.com/ppy/osu/issues/9017, but one thing that's different here is that it also bypasses the star rating filter so I think I'll keep this open?

**Comment by Firmatorenio** [comment]
I'm not sure it's actually related to that, because it seems to me that clicking the hidden difficulties that are hidden because they're out of the star rating filter is an intended behavior too.

The problem here is, the OP wants to play converts of a standard map, but switching difficulties by clicking on their icon keeps switching the mode back to standard.

**Comment by peppy** [comment]
Just to confirm, in this scenario you would expect it to not select and not change ruleset?

**Comment by AnzoP** [comment]
I was expecting to have the ruleset intact when temporarily selecting a difficulty outside the range.
I think I was testing a custom ruleset at that time when I noticed that it changes back to osu!standard ruleset, unless I set the range back to 0 stars.

**Comment by peppy** [comment]
Right, so basically the same as what happens when selecting the icon when you are already in the correct ruleset. Sounds fair.
